gemfirexd/client/src/main/java/com/pivotal/gemfirexd/internal/client/am/Statement.java

```diff
@@ -2715,6 +2715,11 @@ void parseSqlAndSetSqlModes(String sql) throws SqlException {
             firstToken = getStatementToken(sql, idx + 1);
           }
         }
+        if (firstToken.equalsIgnoreCase("set")) {
+          if (sql.toLowerCase().contains(" schema ")) {
+              this.connection_.setSchemaSql_ = sql;
+          }
+        }
 // GemStone changes END
```

Comment:
'contains(" schema ")' will not work correctly in all cases. You could use a regex with " \t\n\r\f" for whitespace instead of a blank. However, still contains can give a misleading match. Better option will be to make a call to "getStatementToken(sql, idx)" again with increased idx. The method getStatementToken can be enhanced to return the end index (say if a non-null int[] is passed in argument) or else can be found separately (which can be a bit tricky though). If we could add int[] argument which is set to index if non-null, then can also use it for <local>/<global> handling that is being done above this which is also inaccurate since it just searches for '>' that can be trouble if such were to be inside some comments for example.